src/app/public/modules/flyout/flyout-adapter.service.ts

```diff
@@ -29,6 +30,25 @@ export class SkyFlyoutAdapterService {
     }
   }
 
+  public setFlexClass(breakpoint: SkyMediaBreakpoints) {
+    const flexEl = document.querySelector('.sky-flyout') as HTMLElement;
```

Comment:
Lets chat about this. I totally get your concern. Just don't love having to pass that ref through the media query service and into this. What would you think about just changing it to look at the host tag `sky-flyout` instead of the class name `.sky-flyout`?

Reply:
Hm, we probably shouldn't be injecting the adapter service into the media query service to begin with. What if the `setBreakpoint` method didn't mess with the adapter at all?

**flyout.component.ts**

```
this.flyoutMediaQueryService.setBreakpoint(width);

const breakpoint = this.flyoutMediaQueryService.current;

this.adapterService.setResponsiveClass(this.elementRef, breakpoint);
```

This might add a few extra steps, but it removes the side effect of messing with the DOM when calling the `SkyFlyoutMediaQueryService.setBreakpoint` method.

Reply:
I've done this and also pulled the logic out into a private method so that the code isn't duplicated over and over.